features/partners.feature

```diff
@@ -10,38 +10,121 @@ Functionaliteit: Huwelijken en geregistreerd partnerschappen van een persoon raa
-      Dan bevat het antwoord GEEN partners
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999992806 |
+      En de persoon heeft een partner met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999992935 |
+      En de partner heeft de volgende naam gegevens
+      | naam              | waarde |
+      | voornamen (02.10) | Osama  |
+      En de partner heeft de volgende aangaanHuwelijkPartnerschap gegevens
+      | naam                      | waarde   |
+      | datum aangaan (06.10)     | 20000115 |
+      | indicatie onjuist (84.10) |          |
+      En de partner heeft de volgende ontbindingHuwelijkPartnerschap gegevens
+      | naam                     | waarde   |
+      | datum ontbinding (07.10) | 20011109 |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                          |
+      | type                | RaadpleegMetBurgerservicenummer |
+      | burgerservicenummer | 999993380                       |
+      | fields              | partners.naam.voornamen         |
+      Dan heeft de persoon met burgerservicenummer '999993380' GEEN 'partners'
 
     Scenario: Een actueel huwelijk en een ontbonden huwelijk
-      Gegeven de persoon met burgerservicenummer 999991553 heeft de volgende huwelijken/partnerschappen in de registratie
-        | Categorie | Voornamen | Datum aangaan (06.10) | Datum ontbinding (07.10) |
-        | 5         | Karel     | 20091102              |                          |
-        | 5         | Björn     |                       | 20080706                 |
-        | 55        | Björn     | 19870714              |                          |
-        | 55        | Björn     | 19870714              |                          |
-      Als de persoon met burgerservicenummer 999991553 wordt geraadpleegd met fields=partners
-      Dan bevat het antwoord 1 partner
-      En bevat het antwoord de partner met naam.voornamen "Karel"
-      En bevat het antwoord GEEN partner met naam.voornamen "Björn"
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999991553 |
+      En de persoon heeft een partner met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999992935 |
+      En de partner heeft de volgende naam gegevens
+      | naam              | waarde |
+      | voornamen (02.10) | Karel  |
+      En de partner heeft de volgende aangaanHuwelijkPartnerschap gegevens
+      | naam                      | waarde   |
+      | datum aangaan (06.10)     | 20091102 |
+      | indicatie onjuist (84.10) |          |
+      En de persoon heeft een partner met de volgende gegevens
+      | naam                | waarde    |
```

Comment:
ook hier zou ik alleen de ontbindingsgegevens noemen. De andere gegevens zitten in de historische categorie.

Reply:
Begrijp ik het goed dat bij een ontbonden relatie dat je een partner krijgt zonder burgerservicenummer en naam gegevens? In dit geval vraag ik mij dan af wat het nut is om het terug te geven? Volgens mij kan een consumer dan niks met die partner. Zeker niet als hij niet eerder een snapshot van die persoon heeft in zijn eigen database

Reply:
Bij een ontbonden relatie zijn er voor die partner ten minste twee voorkomens: 1 categorie 5 (de actuele gegevens) en 1 categorie 55 (de historische gegevens. 

Voor dit scenario zou er in de registratie staan:

| stapel | volgnr | categorie | burgerservicenummer | voornamen | datum aangaan | datum ontbinding |
|--- |--- |--- |--- |--- |--- |--- |
| 1         | 2           | 5              |                                        |                     |                            | 20080706             |
| 1         | 1           | 55           | 999992936                  | Björn           | 19870714           |                                |
| 2        | 1           | 5              | 999992935                  | Karel           | 20091102           |                                |

De API voor personen bevragen levert dan niet stapel 1 (Björn), omdat in de actuele categorie datum ontbinding voorkomt. De API personen bevragen levert alleen huwelijken/partnerschappen die niet ontbonden zijn .
De partner van stapel 2 (Karel) wordt wel geleverd.

Deze feature partners moet m.i. precies dit laten zien, n.l. dat voor elke partner in de registratie moet worden gecontroleerd of dit een "actuele", dus niet ontbonden relatie is. En dat alleen de actuele partners opgenomen moeten worden in het antwoord.

Dus je vraag "In dit geval vraag ik mij dan af wat het nut is om het terug te geven?" is in de feature beantwoord: dat heeft geen nut en dus leveren we het niet.

Reply:
Ik denk dat we niet helemaal hetzelfde beeld hebben hoe ik de functionaliteit probeer te beschrijven.

Ik probeer te beschrijven wat de BRP proxy van de RVIG BRP variant moet krijgen zodat de BRP proxy de mapping kan maken. Jij geeft echter steeds aan dat de gegevens anders zijn vastgelegd in de registratie, maar daar komt de BRP proxy niet bij.

Ik begin mij daarom steeds meer af te vragen of BRP proxy eigenlijk niet tegen de BRP historie API RVIG variant moet praten, omdat er in HC variant een partnerhistorie type voorkomt die zowel categorie 5 als 55 gegevens retourneert die nodig zijn om de ontbonden relatie scenario's te kunnen realiseren.